src/containers/signupFlow/index.tsx

```diff
@@ -1,29 +1,144 @@
+  function convertToYearMonthDateString(d: Date): string {
+    return d.toISOString().split('T')[0];
+  }
+
+  const submitGMForm = async (photoRelease: boolean) => {
+    const profilePicture = await encodeProfileFieldFile(GMForm.profilePicture);
+    dispatch(
+      signup(
+        {
+          email: GMForm.email,
+          password: GMForm.password,
+          firstName: GMForm.firstName,
+          lastName: GMForm.lastName,
+          phoneNumber: GMForm.phoneNumber,
+          location: {
+            address: GMForm.address,
+            city: GMForm.city,
+            state: GMForm.state,
+            zipCode: GMForm.zip,
+          },
+          photoRelease,
+        },
+        {
+          mainContact: {
+            email: GMForm.email,
+            firstName: GMForm.firstName,
+            lastName: GMForm.lastName,
+            dateOfBirth: convertToYearMonthDateString(GMForm.birthday),
+            phoneNumber: GMForm.phoneNumber,
+            pronouns: GMForm.pronouns,
+            allergies: GMForm.allergies ?? null,
+            diagnosis: GMForm.diagnosis ?? null,
+            medications: GMForm.medication ?? null,
+            notes: GMForm.otherNotes ?? null,
+            referrer: GMForm.referrer === 'none' ? null : GMForm.referrer,
+            profilePicture,
+          },
+          additionalContacts: [],
+          children: [],
+        },
+      ),
+    );
   };
-  const submitPFForm = (photoRelease: boolean) => {
-    // temporary console logs, just to show data is being collected correctly.
-    console.log(PFForm1); // tslint:disable-line
-    console.log(PFForm2); // tslint:disable-line
-    console.log(photoRelease); // tslint:disable-line
+
+  const submitPFForm = async (photoRelease: boolean) => {
+    dispatch(
+      signup(
+        {
+          email: PFForm1.email,
+          password: PFForm1.password,
+          firstName: PFForm1.firstName,
+          lastName: PFForm1.lastName,
+          phoneNumber: PFForm1.phoneNumber,
+          location: {
+            address: PFForm1.address,
+            city: PFForm1.city,
+            state: PFForm1.state,
+            zipCode: PFForm1.zip,
+          },
+          photoRelease,
+        },
+        {
+          mainContact: {
+            firstName: PFForm1.firstName,
+            lastName: PFForm1.lastName,
+            dateOfBirth: convertToYearMonthDateString(PFForm1.birthday),
+            phoneNumber: PFForm1.phoneNumber,
+            pronouns: PFForm1.pronouns,
+            allergies: PFForm1.allergies ?? null,
+            diagnosis: PFForm1.diagnosis ?? null,
+            medications: PFForm1.medication ?? null,
+            notes: PFForm1.otherNotes ?? null,
+            profilePicture: await encodeProfileFieldFile(
+              PFForm1.profilePicture,
+            ),
+            email: PFForm1.email,
+            referrer: PFForm1.referrer,
+          },
+          additionalContacts: PFForm2.contacts
```

Comment:
Careful, `[]` is truthy, so you probably want `contacts.?length` which will be undefined, 0, or > 0